Upsell: Update docs and add Best Practices

ayeshakmaz opened the pull request:
New docs for Upsell

ayeshakmaz commented:
@ashleyseto any recommendations for the last Do and Don't visual examples? 

Do
Plan for the timing of your Upsells with new product launches. Try to create different messages for each time an Upsell appears to the user.

Don't
Keep showing the same Upsell once it has been dismissed. Upsells should only appear a maximum of 2 times to the same user, as they have diminishing returns.

ashleyseto commented:
> @ashleyseto any recommendations for the last Do and Don't visual examples?
> 
> Do
> Plan for the timing of your Upsells with new product launches. Try to create different messages for each time an Upsell appears to the user.
> 
> Don't
> Keep showing the same Upsell once it has been dismissed. Upsells should only appear a maximum of 2 times to the same user, as they have diminishing returns.

![Frame 3](https://user-images.githubusercontent.com/2187170/109713390-0074f180-7b56-11eb-849b-1b9599427580.png)

![Frame 4](https://user-images.githubusercontent.com/2187170/109713406-04087880-7b56-11eb-87ad-4e3129b6b4aa.png)


Found some examples from product that I think will work for Do and Don't. It isn't perfect for the visual but I think it gets some of the point across.

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -1,16 +1,30 @@
+    <Upsell
+      title="Give $30, get $60 in ads credit"
+      message="Earn $60 of ads credit, and give $30 of ads credit to a friend"
+      primaryAction={{href: "https://pinterest.com", label: "Send invite"}}
```
nit: Can we add target blank?
`      primaryAction={{href: "https://pinterest.com", target: 'blank', label: "Send invite"}}
`

AlbertCarreras replied:
Same for most of all examples.

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -22,22 +36,16 @@ card(
-          'Text to render inside the Upsell to convey detailed information to the user. The message text has a fixed size.',
-        ],
-        href: '',
+        description: `Main content of Upsell, explains what is being offered or recommended. Content should be [localized](/Localization). See [Best Practices](#Best-practices) for more info.`,
```
typo
`[localized](#Localization)`

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -22,22 +36,16 @@ card(
-        ],
-        href: '',
+        description: `
+        Adds a dismiss button to the Upsell. The \`accessibilityLabel\` should follow the [Accessibility guidelines](#accessibility)
```
typo
[Accessibility guidelines](#Accessibility)

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -46,109 +54,366 @@ card(
         defaultValue: null,
         description: [
-          'Either `<Image />` or `<Icon />` to render on left side of banner. Width is not used with Icon. Image width defaults to 128 px. Max width of image is 128 px.',
+          'Either an [Icon](/icon) or an [Image](/image) to render at the start of the banner. Width is not used with Icon. Image width defaults to 128px. See the [Icon](#Icon) and [Image](#Image) variants for more info.',
```
half typo
`[Icon](/Icon)`

AlbertCarreras replied:
`[Image](/Image)`

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -46,109 +54,366 @@ card(
-          'Heading to render inside the Upsell above the message to convey the Upsell topic to the user. The message text has a fixed size.',
-        ],
-        href: '',
+        description: `Brief title summarizing the Upsell. Content should be [localized](/Localization).`,
```
typo
[localized](#Localization)

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -46,109 +54,366 @@ card(
+        cardSize="lg"
+        type="do"
+        description={`
+        Place the Upsell at the top of the page under the primary navigation when possible.
```
the Upsell > Upsell

AlbertCarreras commented:
Quick comments on DO/DONTs: https://www.loom.com/share/0161fcc0e63340ce921f9a264218bd34

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -46,109 +54,366 @@ card(
+        type="do"
+        description="Use Upsells for marketing new products or encouraging upgrades."
+        defaultCode={`
+          <Upsell
```
indentation

```
<Upsell
  title="Give $30, get $60 in ads credit"
  message="Earn $60 of ads credit, and give $30 of ads credit to a friend"
  primaryAction={{
    href: 'https://pinterest.com',
    label: 'Send invite',
    accessibilityLabel: 'Invite friend to use ads',
  }}
  dismissButton={{
    accessibilityLabel: 'Dismiss banner',
    onDismiss: () => {},
  }}
  imageData={{
    component: <Icon icon="pinterest" accessibilityLabel="" color="darkGray" size={32} />,
  }}
/>;

```

AlbertCarreras replied:
also " vs '

AlbertCarreras replied:
this applies to all examples. I won't copy paste reformating, mostly affects spacing/identation

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -22,133 +41,370 @@ card(
+);
+
+card(
+  <MainSection name="Variants">
```
Any good use case for this? When do we wanna use only-texts Upsells? It defeats the purpose of `promoting an action or upgrading something`. But there might be a good use case.

ayeshakmaz replied:
@ashleyseto do you know of any product examples where Upsell is used without a title or actions? I only have dummy text here right now

ayeshakmaz replied:
(clarified in slack)

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -22,133 +41,370 @@ card(
+
+      For the \`dismissButton\` [IconButton](/IconButton), the label provided should indicate the intent, like “Dismiss this banner”.
+
+      The [Image](/Image) or [Icon](/Icon) supplied to \`imageData\` should only supply an \`alt\` or \`accessibilityLabel\`, respectively, if the Image or Icon supplies extra context or information. Icons in Upsells are often purely decorative, and can therefore have an empty string as the \`accessibilityLabel\`.
```
one of those nits, The Image > Image

ayeshakmaz replied:
I think here it makes sense since we are referring to a particular image or icon, and not the component as a whole

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -158,38 +414,19 @@ card(
+    <MainSection.Subsection
+      title="Actions"
+      description={`
+      Upsells can have either one primary action, or a primary action and a secondary action. These actions can be Buttons, when no \`href\` is supplied, or Links, by specifying the \`href\`  property.
```
`These actions can be buttons, when no \`href\` is supplied, or links, by specifying the \`href\`  property.`

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -158,38 +414,19 @@ card(
-      \`primaryAction\` and \`secondaryAction\` can be used as buttons when no \`href\` is supplied.
-    `}
-    defaultCode={`
+      For example, “Learn more” may link to a separate documentation site, while “Send invite” could be a Button that opens a [Modal](/Modal) with an invite flow. Be sure to localize the labels of the actions.`}
```
`“Send invite” could be a button`

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -22,133 +41,370 @@ card(
+);
+
+card(
+  <MainSection name="Variants">
```
Variants aint ordered by alphabetical order.

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -198,57 +435,111 @@ function Example(props) {
+);
+
+card(
+  <MainSection name="Related">
```
Variants aint ordered by alphabetical order.

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -198,57 +435,111 @@ function Example(props) {
+  <MainSection name="Related">
+    <MainSection.Subsection
+      description={`
+      **[Callout](/callout)**
```
Nit
Callout](/Callout)
[Toast](/Toast)**

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -198,57 +435,111 @@ function Example(props) {
+      Use Callout when communicating critical information, such as an error or warning. Callout can also be used to present the user with general information and further actions they can take, like the successful creation of a business account.
+
+      **[Toast](/toast)**
+      Toast provides feedback on a user interaction, like a confirmation that appears when a Pin has been saved. Unlike Upsell and Callout, Toasts don’t contain actions. They’re also less persistent, and typically disappear after a certain duration.
```
not sure, But i'd say toasts always disappear?

ayeshakmaz commented:
@AlbertCarreras I updated the examples based on Ashley's comment above, I think the loom link might have been outdated, or right before I added those :P

ashleyseto commented:
<img width="668" alt="Screen Shot 2021-03-03 at 4 53 41 PM" src="https://user-images.githubusercontent.com/2187170/109893830-3fcc3c80-7c41-11eb-91f3-b6e689219c61.png">

We could add the error icon here. I'm not sure if there is supposed to be one but it didn't show.

AlbertCarreras commented on docs/src/Upsell.doc.js:
```diff
@@ -22,174 +41,412 @@ card(
+</Box>;
+        `}
+      />
+      <MainSection.Card
```
@ayeshakmaz I'm a little bit confused as this example is not showing on the Docs....

AlbertCarreras replied:
![Uploading image.png…]()

AlbertCarreras commented:
![image](https://user-images.githubusercontent.com/10593890/110006096-372e4180-7ce7-11eb-9a45-751bb1ec0b68.png)

ashleyseto commented:
> ![image](https://user-images.githubusercontent.com/10593890/110006096-372e4180-7ce7-11eb-9a45-751bb1ec0b68.png)
> Same for this:
> ![image](https://user-images.githubusercontent.com/10593890/110006150-4a411180-7ce7-11eb-8cc8-e75ca07a0142.png)
> 
> I read these two stacked Upsells, as 2 isolated Upsells showing in consecutive moments in time?
> 
> Should we add some kind of time hint?
> 
> **_First Upsell. Next Upsell._**
> 
> Maybe we could show the next one after dismissing the first?

We could add text if that helps with clarity. 

I would make the examples though work in the static form rather than having to interact to understand the message.

dangerismycat reviewed (approved):
LGTM! :shipit:
